From 911b37cabfe262b541c3bfed7401691d746c5f80 Mon Sep 17 00:00:00 2001
From: "James Z.M. Gao" <gaozhiming@360.cn>
Date: Wed, 26 Jun 2019 15:59:55 +0800
Subject: [PATCH 1/4] THRIFT-4897 generate test resource for
 thrift-maven-plugin

generate SharedIdl.jar for passing the test case for thrift-maven-plugin
---
 contrib/thrift-maven-plugin/pom.xml           | 43 +++++++++++++++++++
 .../thrift/maven/TestAbstractThriftMojo.java  |  4 +-
 2 files changed, 45 insertions(+), 2 deletions(-)

diff --git a/contrib/thrift-maven-plugin/pom.xml b/contrib/thrift-maven-plugin/pom.xml
index e11fbbf0..524d7eeb 100644
--- a/contrib/thrift-maven-plugin/pom.xml
+++ b/contrib/thrift-maven-plugin/pom.xml
@@ -33,6 +33,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.8.1</version>
         <configuration>
           <source>1.7</source>
           <target>1.7</target>
@@ -48,6 +49,47 @@
           </systemPropertyVariables>
         </configuration>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-jar-plugin</artifactId>
+        <version>3.1.2</version>
+        <executions>
+          <execution>
+            <id>generate-jar-for-test</id>
+            <phase>process-test-classes</phase>
+            <goals>
+              <goal>test-jar</goal>
+            </goals>
+            <configuration>
+              <classifier>idl</classifier>
+              <excludes>
+                <exclude>**/*.class</exclude>
+              </excludes>
+              <includes>
+                <include>**/*.thrift</include>
+              </includes>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>com.coderplus.maven.plugins</groupId>
+        <artifactId>copy-rename-maven-plugin</artifactId>
+        <version>1.0</version>
+        <executions>
+          <execution>
+            <id>rename-jar-for-test</id>
+            <phase>process-test-classes</phase>
+            <goals>
+              <goal>rename</goal>
+            </goals>
+            <configuration>
+              <sourceFile>${project.build.directory}/${project.artifactId}-${project.version}-idl.jar</sourceFile>
+              <destinationFile>${project.build.directory}/SharedIdl.jar</destinationFile>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
     </plugins>
   </build>
   <dependencies>
@@ -88,5 +130,6 @@
     <thrift.root>${basedir}/../..</thrift.root>
     <thrift.compiler>${thrift.root}/compiler/cpp/thrift</thrift.compiler>
     <thrift.test.home>${thrift.root}/test</thrift.test.home>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
   </properties>
 </project>
diff --git a/contrib/thrift-maven-plugin/src/test/java/org/apache/thrift/maven/TestAbstractThriftMojo.java b/contrib/thrift-maven-plugin/src/test/java/org/apache/thrift/maven/TestAbstractThriftMojo.java
index c1176712..7c90a990 100644
--- a/contrib/thrift-maven-plugin/src/test/java/org/apache/thrift/maven/TestAbstractThriftMojo.java
+++ b/contrib/thrift-maven-plugin/src/test/java/org/apache/thrift/maven/TestAbstractThriftMojo.java
@@ -59,7 +59,7 @@ public class TestAbstractThriftMojo {
         // used by other tests. It's used here to represent a dependency of the project maven is building,
         // one that is contributing .thrift IDL files as well as any other artifacts.
         final Iterable<File> classpathElementFiles = Lists.newArrayList(
-                new File("src/test/resources/dependency-jar-test/SharedIdl.jar")
+                new File("target/SharedIdl.jar")
         );
 
         final Set<File> thriftDirectories = mojo.makeThriftPathFromJars(temporaryThriftFileDirectory, classpathElementFiles);
@@ -70,7 +70,7 @@ public class TestAbstractThriftMojo {
         // means it points to the directory containing the "idl" hierarchy rather than to the idl directory
         // itself.
         final Set<File> expected = Sets.newHashSet(
-                new File(testRootDir, "src/test/resources/dependency-jar-test/SharedIdl.jar")
+                new File(testRootDir, "target/SharedIdl.jar")
         );
 
         assertEquals("makeThriftPathFromJars should return thrift IDL base path from within JAR", expected, thriftDirectories);
-- 
2.22.0

